.github/workflows/ci.yml

```
@@ -37,3 +37,31 @@
       - name: Generate code coverage report
         if: always()
         uses: davelosert/vitest-coverage-report-action@v2
+
+      - name: Build project
+        run: npm run build
+
+      - name: Upload production-ready build files
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        uses: actions/upload-artifact@v2
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)